File: Leetcode/0819-most-common-word/0819-most-common-word.java
class Solution {
    public String mostCommonWord(String paragraph, String[] banned) {
        Map<String, Integer> counts = new HashMap<>();

        String[] texts = paragraph.replaceAll("[!?',;.]", " ").toLowerCase().split(" ");

        for(String text : texts) {
            if(text == "" || Arrays.asList(banned).indexOf(text) >= 0) {
                continue;
            }
            counts.put(text, counts.getOrDefault(text, 0) + 1);
        }

        return Collections.max(counts.entrySet(), Map.Entry.comparingByValue()).getKey();
    }
}

File: Leetcode/0819-most-common-word/0819-most-common-word.kt
class Solution {
    fun mostCommonWord(paragraph: String, banned: Array<String>): String {
        val counts : MutableMap<String, Int> = mutableMapOf()

        val words = paragraph.replace("\\W+".toRegex(), " ").toLowerCase().trim().split(" ");

        for(word in words) {
            if(!banned.contains(word)) {
                counts[word] = counts.getOrDefault(word, 0) + 1
            }
        }

        return counts.maxBy { it.value }!!.key
    }
}